packages/test/test-end-to-end-tests/src/test/SummarizerWithDataStoreRealization.spec.ts

```diff
@@ -0,0 +1,329 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import {
+    IAckedSummary,
+    IContainerRuntimeOptions,
+    ISummarizer,
+} from "@fluidframework/container-runtime";
+import { IFluidHandle, IRequest } from "@fluidframework/core-interfaces";
+import { FluidDataStoreRuntime, mixinSummaryHandler } from "@fluidframework/datastore";
+import { SharedMatrix } from "@fluidframework/matrix";
+import { SharedMap } from "@fluidframework/map";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider,
+    wrapDocumentServiceFactory,
+    waitForContainerConnection,
+    summarizeNow,
+    createSummarizerFromFactory,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, getContainerRuntimeApi } from "@fluidframework/test-version-utils";
+import { IContainerRuntimeBase, IFluidDataStoreFactory } from "@fluidframework/runtime-definitions";
+import { ISummaryContext } from "@fluidframework/driver-definitions";
+import { ISummaryTree } from "@fluidframework/protocol-definitions";
+import { UndoRedoStackManager } from "@fluidframework/undo-redo";
+import { SharedCounter } from "@fluidframework/counter";
+import { pkgVersion } from "../packageVersion";
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type ProviderPropertyKeys<T extends Object, TProp extends keyof T = keyof T> = string extends TProp
```

Comment:
The search related code is not needed. Please look at my comment in the the mixinSummaryHandler.